src/lib/containers/entity_finder/tree/TreeView.tsx

```diff
@@ -0,0 +1,253 @@
+        )
+
+        break
+      case FinderScope.FAVORITES: {
+        SynapseClient.getUserFavorites(sessionToken).then(({ results }) => {
+          setTopLevelEntities(results)
+          setIsLoading(false)
+        })
+        break
+      }
+      case FinderScope.CURRENT_PROJECT:
+      default:
+        SynapseClient.getEntityPath(sessionToken, initialContainer).then(
+          path => {
+            setInitialContainerPath(path)
+            setTopLevelEntities([path.path[1]])
+            setIsLoading(false)
+          },
+        )
+
+        break
+    }
+  }, [sessionToken, scope, initialContainer])
+
+  useEffect(() => {
+    if (currentContainer === 'root') {
+      switch (scope) {
+        case FinderScope.ALL_PROJECTS:
+          setDetailsViewConfiguration({
+            type: EntityDetailsListDataConfigurationType.USER_PROJECTS,
+          })
+          break
+        case FinderScope.CREATED_BY_ME:
+          setDetailsViewConfiguration({
+            type: EntityDetailsListDataConfigurationType.USER_PROJECTS,
+            getProjectParams: {
+              filter: 'CREATED',
+            },
+          })
+          break
+        case FinderScope.CURRENT_PROJECT:
+          setDetailsViewConfiguration({
+            type: EntityDetailsListDataConfigurationType.HEADER_LIST,
+            headerList: topLevelEntities,
+          })
+          break
+        case FinderScope.FAVORITES:
+          setDetailsViewConfiguration({
+            type: EntityDetailsListDataConfigurationType.USER_FAVORITES,
+            headerList: topLevelEntities,
+          })
+
+          break
+      }
+    } else {
+      setDetailsViewConfiguration({
+        type: EntityDetailsListDataConfigurationType.PARENT_CONTAINER,
+        parentContainerParams: {
+          parentContainerId: currentContainer,
+        },
+      })
+    }
+  }, [scope, currentContainer, topLevelEntities, setDetailsViewConfiguration])
+
+  return (
+    <div className="EntityFinderTreeView" style={{ height: '500px' }}>
+      {/* <div className={`EntityFinderTreeView__SelectionHeader`}></div> */}
+      <div style={{ overflow: 'auto' }}>
+        {isLoading ? (
+          <div className="spinner" />
+        ) : (
+          <>
+            {showFakeRootNode && (
+              <div
+                style={{ paddingLeft: `5px` }}
+                className={`EntityFinderTreeView__Row${
+                  currentContainer === 'root'
+                    ? ' EntityFinderTreeView__Row__Selected'
+                    : ''
+                }`}
+                onClick={() => {
+                  setCurrentContainer('root')
+                }}
+              >
+                <div
+                  className={'EntityFinderTreeView__Row__ExpandButton'}
+                  onClick={e => {
+                    e.stopPropagation()
+                    setExpandFakeRoot(!expandFakeRoot)
+                  }}
+                >
+                  {expandFakeRoot ? '▾' : '▸'}
```

Comment:
expand/collapsed characters/icons should probably be shared with TreeViewRow (may want to update)